go/arrow/array/binarybuilder.go

```diff
@@ -289,6 +289,26 @@ func (b *BinaryBuilder) appendNextOffset() {
 	b.appendOffsetVal(numBytes)
 }
 
+func (b *BinaryBuilder) AppendValueFromString(s string) error {
+	if s == NullValueStr {
+		b.AppendNull()
+		return nil
+	}
+	switch b.dtype.ID() {
+	case arrow.BINARY, arrow.LARGE_BINARY:
+		decodedVal, err := base64.StdEncoding.DecodeString(s)
+		if err != nil {
+			return fmt.Errorf("could not decode base64 string: %w", err)
+		}
```

Comment:
do we want to try falling back to `RawStdEncoding` if this errors so we can handle both situations of padding or no padding?

Reply:
hmm. I think we should try to always use `std` otherwise this fallback will propagate everywhere because we want catch if someone encoded something with `RawStdEncoding` instead of `StdEncoding`.

Reply:
that's fair. I suppose if we run into an issue for individuals that are using non-padded base64 for this they'll file an issue and it's fairly easy to add the support. so this is fine.